File: 3083-existence-of-a-substring-in-a-string-and-its-reverse/3083-existence-of-a-substring-in-a-string-and-its-reverse.java
class Solution {
    public boolean isSubstringPresent(String s) {
        List<String> nm=new ArrayList<>();
        for(int i=0;i<s.length()-1;i++)
        {
            nm.add(s.charAt(i)+""+s.charAt(i+1));
        }
        String rev=new StringBuilder(s).reverse().toString();
        for(int i=0;i<nm.size();i++)
        {
            if(s.contains(nm.get(i)) && rev.contains(nm.get(i)))
            {
                return true;
            }
        }
        return false;
    }
}

File: 3083-existence-of-a-substring-in-a-string-and-its-reverse/3083-existence-of-a-substring-in-a-string-and-its-reverse.cpp
class Solution {
public:
  bool isSubstringPresent(string s) {
    for (int i = 0; i < s.size() - 1; i++) {
    
      string substring = s.substr(i, 2);

    
      reverse(substring.begin(), substring.end());

      
      if (s.find(substring) != string::npos) {
        return true;
      }
    }


    return false;
  }
};
auto init = []() {
    ios::sync_with_stdio(false);
    cin.tie(nullptr);
    cout.tie(nullptr);
    return 'c';
}();